pg_shard.c

```diff
@@ -586,6 +607,25 @@ SelectFromMultipleShards(Query *query, List *queryShardList)
 
 
 /*
+ * TODO: Add documentation
+ */
+static Query *
+BuildLocalQuery(Query *query, List *localRestrictList)
```

Comment:
It doesn't modify the input, it returns a modified copy (this ties into the question about `copyObject` later).